src/pudl_rmi/connect_ferc1_to_eia.py

```diff
@@ -47,30 +47,34 @@
 from sklearn.model_selection import KFold  # , cross_val_score
 
 import pudl_rmi
-from pudl_rmi import connect_deprish_to_eia, make_plant_parts_eia
+from pudl_rmi import connect_deprish_to_eia
 
 logger = logging.getLogger(__name__)
 # Silence the recordlinkage logger, which is out of control
 logging.getLogger("recordlinkage").setLevel(logging.ERROR)
 IDX_STEAM = ["utility_id_ferc1", "plant_id_ferc1", "report_date"]
 
 
-def execute(pudl_out, plant_parts_eia):
+def execute(train_df, pudl_out, plant_parts_eia_distinct):
     """
     Coordinate the connection between FERC1 plants and EIA plant-parts.
 
+    Args:
+        train_df (pandas.DataFrame): Training data connected to EIA
+        plant-parts data such that a FERC1 plant record is reported at the same
+        granularity as the connected EIA plant-parts record. MultiIndex
+        with record_id_eia and record_id_ferc1.
     Note: idk if this will end up as a script or what, but I wanted a place to
     coordinate the connection. May be temporary.
     """
-    inputs = InputManager(pudl_rmi.TRAIN_FERC1_EIA_CSV, pudl_out, plant_parts_eia)
+    inputs = InputManager(train_df, pudl_out, plant_parts_eia_distinct)
```

Comment:
Previously when the `InputManager` was taking `plant_parts_eia` as input it was necessary to have both the full PPL and the distinct PPL loaded in memory at the same time. The full PPL was just needed to connect the training data with true granularity EIA records so I moved this whole process to before this `connect_ferc1_to_eia.execute` function. The result of that is `train_df` which I pass directly to `InputManager`. Additionally I create and save the distinct PPL optionally during the PPL processing so the distinct PPL can be read directly without needing to also load the full PPL.